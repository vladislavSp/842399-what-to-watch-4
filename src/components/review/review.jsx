import React from "react";
import PropTypes from "prop-types";
import SignHeader from "../sign-in/header/sign-header.jsx";
import withReviewState from "../../hocs/with-review-state.jsx";
import {Link} from "react-router-dom";

export const Review = (props) => {
  const {
    currentMovie,
    networkError,
    onSubmit,
    onRatingChange,
    onReviewChange,
    reviewIsValid,
    ratingIsValid,
    isLoading
  } = props;
  const {title, background, backgroundColor, posterImage} = currentMovie;

  return (
    <section style={{backgroundColor}} className="movie-card movie-card--full">
      <div className="movie-card__header">
        <div className="movie-card__bg">
          <img src={background} alt={title} />
        </div>

        <h1 className="visually-hidden">WTW</h1>

        <header className="page-header">
          <div className="logo">
            <Link to="/" className="logo__link">
              <span className="logo__letter logo__letter--1">W</span>
              <span className="logo__letter logo__letter--2">T</span>
              <span className="logo__letter logo__letter--3">W</span>
            </Link>
          </div>

          <nav className="breadcrumbs">
            <ul className="breadcrumbs__list">
              {/* link on film */}
              <li className="breadcrumbs__item">
                <Link
                  to={`/films/${currentMovie.id}`}
                  className="breadcrumbs__link"
                >
                  {title}
                </Link>
              </li>
              <li className="breadcrumbs__item">
                <a className="breadcrumbs__link">Add review</a>
              </li>
            </ul>
          </nav>

          <SignHeader />
        </header>

        <div className="movie-card__poster movie-card__poster--small">
          <img
            src={posterImage}
            alt={title + `poster`}
            width="218"
            height="327"
          />
        </div>
      </div>

      <div className="add-review">
        {networkError && (
          <div className="sign-in__message">
            <p>Error occured {networkError}</p>
          </div>
        )}

        <form onSubmit={onSubmit} action="#" className="add-review__form">
          <div className="rating">
            <div className="rating__stars">
              <input
                onChange={onRatingChange}
                className="rating__input"
                id="star-1"
                type="radio"
                name="rating"
                value="1"
              />
              <label className="rating__label" htmlFor="star-1">
                Rating 1
              </label>

              <input
                onChange={onRatingChange}
                className="rating__input"
                id="star-2"
                type="radio"
                name="rating"
                value="2"
              />
              <label className="rating__label" htmlFor="star-2">
                Rating 2
              </label>

              <input
                onChange={onRatingChange}
                className="rating__input"
                id="star-3"
                type="radio"
                name="rating"
                value="3"
                defaultChecked
              />
              <label className="rating__label" htmlFor="star-3">
                Rating 3
              </label>

              <input
                onChange={onRatingChange}
                className="rating__input"
                id="star-4"
                type="radio"
                name="rating"
                value="4"
              />
              <label className="rating__label" htmlFor="star-4">
                Rating 4
              </label>

              <input
                onChange={onRatingChange}
                className="rating__input"
                id="star-5"
                type="radio"
                name="rating"
                value="5"
              />
              <label className="rating__label" htmlFor="star-5">
                Rating 5
              </label>
            </div>
          </div>

          <div className="add-review__text">
            <textarea
              onChange={onReviewChange}
              disabled={isLoading}
              className="add-review__textarea"
              name="review-text"
              id="review-text"
              placeholder="Review text"
            ></textarea>

            <div className="add-review__submit">
              <button
                disabled={(!reviewIsValid && ratingIsValid) || isLoading}
                className="add-review__btn"
                type="submit"
              >
                Post
              </button>
            </div>
          </div>
        </form>
      </div>
    </section>
  );
};

Review.propTypes = {
  currentMovie: PropTypes.object.isRequired,
  networkError: PropTypes.bool,
  onSubmit: PropTypes.func,
  onRatingChange: PropTypes.func,
  onReviewChange: PropTypes.func,
  reviewIsValid: PropTypes.bool,
  ratingIsValid: PropTypes.bool,
  isLoading: PropTypes.bool
};

export default withReviewState(Review);
