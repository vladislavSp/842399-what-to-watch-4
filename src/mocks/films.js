const films = [
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Fantastic Beasts`,
    genre: `Comedies`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Bohemian Rhapsody`,
    genre: `Documentary`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Macbeth`,
    genre: `Comedies`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Aviator`,
    genre: `Dramas`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `We need to talk about Kevin`,
    genre: `Thrillers`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Shutter Island`,
    genre: `Horror`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `No Country for Old Men`,
    genre: `Romance`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Snatch`,
    genre: `Sci-Fi`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  }
];

const promoFilm = {
  promoTitle: `The Grand Budapest Hotel`,
  promoGenre: `Comedy/Drama`,
  promoYear: 2014,
};

export {films, promoFilm};
