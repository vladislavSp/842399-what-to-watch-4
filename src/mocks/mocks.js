const films = [
  {
    id: 0,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 1,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 2,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 3,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 4,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 5,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 6,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 7,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  }
];

const promoFilm = {
  id: `1`,
  title: `The Grand Budapest Hotel`,
  genre: `Comedy/Drama`,
  year: 2014,
  posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Snatch.jpg`,
  previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/snatch.jpg`
};

const filmContent = {
  title: `Gangs of new york`,
  producer: `Martin Scorsese`,
  genre: `Crime`,
  year: 2002,
  rating: `8.8`,
  actors: [
    `Leonardo DiCaprio`,
    `Cameron Diaz`,
    `Daniel Day-Lewis`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`
  ],
  description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
  posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
  previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
  background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
  video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
};

const reviews = [
  {
    id: 0,
    text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    author: `Kate Muir`,
    date: `December 24, 2016`,
    rating: `8,9`
  },
  {
    id: 1,
    text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    author: `Kate Muir`,
    date: `December 24, 2016`,
    rating: `8,9`
  }
];

export const userObj = {
  id: `1`,
  avatarUrl: `avatar`,
  email: `email`,
  name: `name`
};

export {films, promoFilm, filmContent, reviews};
