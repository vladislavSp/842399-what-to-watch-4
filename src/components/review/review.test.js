import React from 'react';
import renderer from 'react-test-renderer';
import {Review} from './review.jsx';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {Router} from 'react-router-dom';
import history from '../../history';
import {films} from '../../mocks/mocks';

const mockStore = configureStore([]);

const store = mockStore({
  APP: {
    activeGenre: `All genre`,
    filmLength: 8
  },
  USER: {
    authorizationStatus: `NO_AUTH`,
    authorizationError: false,
    user: {id: 1, avatar: `avatar`, email: `mail`, name: `name`}
  }
});

it(`Review render correctly`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <Provider store={store}>
            <Review
              currentMovie={films[0]}
            />
          </Provider>
        </Router>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
