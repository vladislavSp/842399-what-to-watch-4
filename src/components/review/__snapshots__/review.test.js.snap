// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review render correctly 1`] = `
<section
  className="movie-card movie-card--full"
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
>
  <div
    className="movie-card__header"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Gangs of new york"
        src="https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <nav
        className="breadcrumbs"
      >
        <ul
          className="breadcrumbs__list"
        >
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
              href="movie-page.html"
            >
              Gangs of new york
            </a>
          </li>
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
            >
              Add review
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="user-block"
      >
        <a
          className="user-block__link"
          href="/login"
          onClick={[Function]}
        >
          Sign in
        </a>
      </div>
    </header>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        alt="Gangs of new yorkposter"
        height="327"
        src="https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg"
        width="218"
      />
    </div>
  </div>
  <div
    className="add-review"
  >
    <form
      action="#"
      className="add-review__form"
    >
      <div
        className="rating"
      >
        <div
          className="rating__stars"
        >
          <input
            className="rating__input"
            id="star-1"
            name="rating"
            type="radio"
            value="1"
          />
          <label
            className="rating__label"
            htmlFor="star-1"
          >
            Rating 1
          </label>
          <input
            className="rating__input"
            id="star-2"
            name="rating"
            type="radio"
            value="2"
          />
          <label
            className="rating__label"
            htmlFor="star-2"
          >
            Rating 2
          </label>
          <input
            className="rating__input"
            defaultChecked={true}
            id="star-3"
            name="rating"
            type="radio"
            value="3"
          />
          <label
            className="rating__label"
            htmlFor="star-3"
          >
            Rating 3
          </label>
          <input
            className="rating__input"
            id="star-4"
            name="rating"
            type="radio"
            value="4"
          />
          <label
            className="rating__label"
            htmlFor="star-4"
          >
            Rating 4
          </label>
          <input
            className="rating__input"
            id="star-5"
            name="rating"
            type="radio"
            value="5"
          />
          <label
            className="rating__label"
            htmlFor="star-5"
          >
            Rating 5
          </label>
        </div>
      </div>
      <div
        className="add-review__text"
      >
        <textarea
          className="add-review__textarea"
          id="review-text"
          name="review-text"
          placeholder="Review text"
        />
        <div
          className="add-review__submit"
        >
          <button
            className="add-review__btn"
            type="submit"
          >
            Post
          </button>
        </div>
      </div>
    </form>
  </div>
</section>
`;
